File: src/#817/solution.js
function ListNode(val, next) {
  this.val = val === undefined ? 0 : val
  this.next = next === undefined ? null : next
}

var numComponents = function (head, G) {

  let Gmap = new Map()

  G.forEach(num => {
    Gmap.set(num, true)
  })

  let res = 0
  let check = false
  let current = head

  while (current !== null) {
    if (Gmap.has(current.val)) {
      if (!check) {
        res++
        check = true
      }
    } else {
      check = false
    }
    current = current.next
  }
  
  return res
}

File: src/#817/solution.go
package solution

type ListNode struct {
	Val  int
	Next *ListNode
}

func numComponents(head *ListNode, G []int) int {

	Gmap := map[int]bool{}

	for _, n := range G {
		Gmap[n] = true
	}

	res := 0
	check := false
	current := head

	for current != nil {
		if ok := Gmap[current.Val]; ok {
			if !check {
				res++
				check = true
			}
		} else {
			check = false
		}
		current = current.Next
	}
	return res
}
